src/synth/fluid_synth_polymono.c

```diff
@@ -0,0 +1,536 @@
+	{
+	    int r =fluid_synth_set_basic_channel_LOCAL( synth, 
+				basicChannelInfos[i].basicchan,
+				basicChannelInfos[i].mode,
+				basicChannelInfos[i].val);
+	    if (result == FLUID_OK) result = r;
+	}
+    }
+    else result = fluid_synth_set_basic_channel_LOCAL( synth, 0, FLUID_CHANNEL_MODE_OMNION_POLY,0);
+    FLUID_API_RETURN(result);
+}
+
+/**
+ * Changes the mode of an existing basic channel or inserts a new basic channel part.
+ *
+ * - If basicchan is already a basic channel, the mode is changed.
+ * - If basicchan is not a basic channel, a new basic channel part is inserted between
+ * the previous basic channel and the next basic channel. val value of the previous
+ * basic channel will be narrowed if necessary. 
+ * 
+ * @param synth the synth instance.
+ * @param basicchan the Basic Channel number (0 to MIDI channel count-1).
+ * @param mode the MIDI mode to use for basicchan (0 to 3).
+ * @param val Number of monophonic channels (for mode 3 only) (0 to MIDI channel count).
+ * 
+ * @return 
+ * - FLUID_OK on success.
+ * - FLUID_FAILED
+ *   - synth is NULL.
+ *   - chan or val is outside MIDI channel count.
+ *   - mode is invalid.
+ *   - val has a number of channels overlapping the next basic channel.
+ */
+int fluid_synth_set_basic_channel(fluid_synth_t* synth, int basicchan, int mode, int val)
+{
+    int result;
+    const int chan = basicchan;
+
+    fluid_return_val_if_fail (mode >= 0, FLUID_FAILED);
+    fluid_return_val_if_fail (mode < FLUID_CHANNEL_MODE_LAST, FLUID_FAILED);
+    fluid_return_val_if_fail (val >= 0, FLUID_FAILED);
+    FLUID_API_ENTRY_CHAN(FLUID_FAILED);
+    /**/
+    if (basicchan + val > synth->midi_channels)
+        FLUID_API_RETURN(FLUID_FAILED);
+
+    result = fluid_synth_set_basic_channel_LOCAL(synth, basicchan,mode,val);
+    /**/
+    FLUID_API_RETURN(result);
+}
+
+/**
+ * Changes the mode of an existing basic channel or inserts a new basic channel part.
+ *
+ * - If basicchan is already a basic channel, the mode is changed.
+ * - If basicchan is not a basic channel, a new basic channel part is inserted between
+ * the previous basic channel and the next basic channel. val value of the previous
+ * basic channel will be narrowed if necessary. 
+ * 
+ * The function is used internally by API fluid_synth_reset_basic_channels() and
+ * fluid_synth_set_basic_channel().
+ * 
+ * @param synth the synth instance.
+ * @param basicchan the Basic Channel number (0 to MIDI channel count-1).
+ * @param mode the MIDI mode to use for basicchan (0 to 3).
+ * @param val Number of monophonic channels (for mode 3 only) (0 to MIDI channel count).
+ * 
+ * @return 
+ * - FLUID_OK on success.
+ * - FLUID_FAILED
+ *   - basicchan is outside MIDI channel count.
+ *   - val has a number of channels overlapping the next basic channel.
+ */
+char * warning_msg1 = "Basic channel %d has been narrowed to %d channels.";
+char * warning_msg2 = "Basic channel %d have number of channels that overlaps.\n\
```

Comment:
`static const` and inside function scope, please.

Reply:
Ok